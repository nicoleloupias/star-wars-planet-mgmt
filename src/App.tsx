import { ChakraProvider } from "@chakra-ui/react";
import { RouterProvider } from "react-router-dom";
import { router } from "./router";
import { theme } from "./theme";
import { useEffect } from "react";
import { usePlanetsStore } from "./hooks/usePlanetsStore";
import { PlanetService } from "./services/PlanetService";

function App() {
  const planets = usePlanetsStore((state) => state.planets);
  const setPlanets = usePlanetsStore((state) => state.setPlanets);

  const getAllPlanets = async () => {
    const planets = await PlanetService.getAll();
    setPlanets(planets);
  };

  useEffect(() => {
    if (!planets) {
      getAllPlanets();
    }
  }, []);

  return (
    <ChakraProvider theme={theme}>
      <RouterProvider router={router} />
    </ChakraProvider>
  );
}

export default App;
