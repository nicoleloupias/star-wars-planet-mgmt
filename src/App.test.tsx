import { describe, expect, test } from "vitest";
import App from "./App";
import { render } from "@testing-library/react";

describe("<App/>", () => {
  test("should render succesfully", () => {
    const component = render(<App />);

    expect(component).toMatchSnapshot();
  });

  // test.only("should get all planets", async () => {
  //   usePlanetsStore.setState((state) => ({ ...state, planets: undefined }));
  //   expect(usePlanetsStore.getState().planets).toBe(undefined);

  //   const component = render(<App />);

  //   await waitFor(() => {
  //     console.log(usePlanetsStore.getState().planets);
  //     // expect(usePlanetsStore.getState().planets?.length).toBe(2);
  //     expect(component.getAllByTestId("planet-card")).toBe(2);
  //   });
  // });
});
