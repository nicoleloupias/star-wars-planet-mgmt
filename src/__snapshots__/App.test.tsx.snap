// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<App/> > should render succesfully 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-dark"
  >
    <div>
      <div
        class="css-0"
      >
        <nav
          class="css-m25t9"
        >
          <a
            aria-current="page"
            class="chakra-link css-spn4bz active"
            href="/"
          >
            Home
          </a>
          <a
            class="chakra-link css-spn4bz"
            href="/create"
          >
            Create
          </a>
        </nav>
        <section
          class="css-1lbat96"
        >
          <div
            class="css-ybip2u"
          />
          <h2
            class="chakra-heading css-s2chw8"
          >
            Explore the galaxy
          </h2>
          <div
            class="css-1etnvhi"
          />
          <p
            class="chakra-text css-1is01ss"
          >
            Scroll
            <svg
              class="chakra-icon css-1j5r502"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                fill="currentColor"
              />
            </svg>
          </p>
          <img
            class="chakra-image css-odqkwc"
            src="/src/assets/Darth-Vader-PNG-HD-Image.png"
          />
        </section>
        <div
          class="css-b93e9j"
        >
          <section
            class="css-1bb2txy"
          >
            <form
              class="css-1654nw7"
            >
              <div
                class="css-nu2kwa"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  role="group"
                >
                  <div
                    class="chakra-input__group css-1y0e7gb"
                    data-group="true"
                  >
                    <input
                      class="chakra-input css-1rrwde8"
                      data-testid="search"
                      id="search"
                      name="search"
                      placeholder="Search"
                      value=""
                    />
                    <div
                      class="chakra-input__right-element css-1lds0jh"
                    >
                      <div
                        class="css-1rs9p06"
                      >
                        <button
                          aria-label="Search"
                          class="chakra-button css-1ni11ii"
                          type="submit"
                        >
                          <svg
                            aria-hidden="true"
                            class="chakra-icon css-onkibi"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                              fill="currentColor"
                            />
                          </svg>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="chakra-select__wrapper css-awd3x6"
              >
                <select
                  class="chakra-select css-szixr"
                  name="sortBy"
                >
                  <option
                    value=""
                  >
                    Order by
                  </option>
                  <option
                    value="name"
                  >
                    Name
                  </option>
                  <option
                    value="diameter"
                  >
                    Diameter
                  </option>
                  <option
                    value="climate"
                  >
                    Climate
                  </option>
                  <option
                    value="terrain"
                  >
                    Terrain
                  </option>
                  <option
                    value="population"
                  >
                    Population
                  </option>
                </select>
                <div
                  class="chakra-select__icon-wrapper css-iohxn1"
                >
                  <svg
                    aria-hidden="true"
                    class="chakra-select__icon"
                    focusable="false"
                    role="presentation"
                    style="width: 1em; height: 1em; color: currentColor;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </form>
            <div
              class="css-16138za"
            />
          </section>
        </div>
      </div>
      <span
        hidden=""
        id="__chakra_env"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <div
        aria-label="Notifications-top"
        aria-live="polite"
        id="chakra-toast-manager-top"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-label="Notifications-top-left"
        aria-live="polite"
        id="chakra-toast-manager-top-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-top-right"
        aria-live="polite"
        id="chakra-toast-manager-top-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-bottom-left"
        aria-live="polite"
        id="chakra-toast-manager-bottom-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-bottom"
        aria-live="polite"
        id="chakra-toast-manager-bottom"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-label="Notifications-bottom-right"
        aria-live="polite"
        id="chakra-toast-manager-bottom-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="css-0"
    >
      <nav
        class="css-m25t9"
      >
        <a
          aria-current="page"
          class="chakra-link css-spn4bz active"
          href="/"
        >
          Home
        </a>
        <a
          class="chakra-link css-spn4bz"
          href="/create"
        >
          Create
        </a>
      </nav>
      <section
        class="css-1lbat96"
      >
        <div
          class="css-ybip2u"
        />
        <h2
          class="chakra-heading css-s2chw8"
        >
          Explore the galaxy
        </h2>
        <div
          class="css-1etnvhi"
        />
        <p
          class="chakra-text css-1is01ss"
        >
          Scroll
          <svg
            class="chakra-icon css-1j5r502"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              fill="currentColor"
            />
          </svg>
        </p>
        <img
          class="chakra-image css-odqkwc"
          src="/src/assets/Darth-Vader-PNG-HD-Image.png"
        />
      </section>
      <div
        class="css-b93e9j"
      >
        <section
          class="css-1bb2txy"
        >
          <form
            class="css-1654nw7"
          >
            <div
              class="css-nu2kwa"
            >
              <div
                class="chakra-form-control css-1kxonj9"
                role="group"
              >
                <div
                  class="chakra-input__group css-1y0e7gb"
                  data-group="true"
                >
                  <input
                    class="chakra-input css-1rrwde8"
                    data-testid="search"
                    id="search"
                    name="search"
                    placeholder="Search"
                    value=""
                  />
                  <div
                    class="chakra-input__right-element css-1lds0jh"
                  >
                    <div
                      class="css-1rs9p06"
                    >
                      <button
                        aria-label="Search"
                        class="chakra-button css-1ni11ii"
                        type="submit"
                      >
                        <svg
                          aria-hidden="true"
                          class="chakra-icon css-onkibi"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                            fill="currentColor"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="chakra-select__wrapper css-awd3x6"
            >
              <select
                class="chakra-select css-szixr"
                name="sortBy"
              >
                <option
                  value=""
                >
                  Order by
                </option>
                <option
                  value="name"
                >
                  Name
                </option>
                <option
                  value="diameter"
                >
                  Diameter
                </option>
                <option
                  value="climate"
                >
                  Climate
                </option>
                <option
                  value="terrain"
                >
                  Terrain
                </option>
                <option
                  value="population"
                >
                  Population
                </option>
              </select>
              <div
                class="chakra-select__icon-wrapper css-iohxn1"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-select__icon"
                  focusable="false"
                  role="presentation"
                  style="width: 1em; height: 1em; color: currentColor;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </form>
          <div
            class="css-16138za"
          />
        </section>
      </div>
    </div>
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
